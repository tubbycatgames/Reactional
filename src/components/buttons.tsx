import React from 'react'

import { t } from '../utils/localization'

import { SchemePrimaryButton } from './schemed'

export const MenuButton = ({ navigate }: any) => (
  <SchemePrimaryButton
    accessibilityLabel={t('buttons.menu.label')}
    onPress={() => navigate('Menu')}
    title={t('buttons.menu.title')}
  />
)
