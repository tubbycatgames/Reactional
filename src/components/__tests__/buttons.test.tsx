import React from 'react'
import renderer from 'react-test-renderer'
import { Button } from 'react-native'

import { MenuButton } from '../buttons'

describe('MenuButton', () => {
  it('should render', () => {
    const render = renderer.create(<MenuButton navigate={() => {}} />)
    expect(render.toJSON()).toMatchSnapshot()
  })

  it('should bind a nav action on button press', () => {
    const navigate = jest.fn()
    const render = renderer.create(<MenuButton navigate={navigate} />)

    const button = render.root.findByType(Button)
    button.props.onPress()
    expect(navigate).toHaveBeenCalledWith('Menu')
  })
})
