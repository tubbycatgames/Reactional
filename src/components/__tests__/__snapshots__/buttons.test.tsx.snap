// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton should render 1`] = `
<View
  accessibilityLabel="Click to return to the Menu"
  accessibilityRole="button"
  accessibilityStates={Array []}
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "padding": 8,
            "textAlign": "center",
          },
          Object {
            "color": "#00CFFF",
          },
        ]
      }
    >
      Menu
    </Text>
  </View>
</View>
`;
